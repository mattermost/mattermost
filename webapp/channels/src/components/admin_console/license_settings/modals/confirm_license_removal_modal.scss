.ConfirmLicenseRemovalModal {
    .modal-header {
        button.close {
            span {
                color: rgba(63, 67, 80, 0.75) !important;
            }
        }
    }

    .modal-body {
        padding-top: 0 !important;
        padding-bottom: 40px !important;
    }

    .content-body {
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;

        .title {
            margin: 10px 0;
            color: var(--sys-denim-center-channel-text);
            font-family: Metropolis;
            font-size: 22px;
            font-weight: 700;
            line-height: 28px;
        }

        .subtitle {
            width: 90%;
            color: var(--sys-denim-center-channel-text);
            font-size: 14px;
            font-style: normal;
            font-weight: normal;
            line-height: 20px;
            text-align: center;
        }

        .alert-svg {
            margin-bottom: 15px;
        }
    }

    .content-footer {
        display: flex;
        align-items: center;
        justify-content: center;
        margin-top: 20px;

        button {
            height: 40px;
            border-radius: 4px;
            margin-top: 15px !important;
            margin-left: 10px;
        }
    }

    .modal-footer {
        border-top: none !important;
        background: none !important;
    }
}
