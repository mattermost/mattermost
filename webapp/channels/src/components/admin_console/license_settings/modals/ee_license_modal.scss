.EELicenseModal {
    .modal-header {
        background: none !important;

        button.close {
            span {
                color: rgba(63, 67, 80, 0.75) !important;
            }
        }
    }

    .title {
        margin: 10px 0;
        color: #3f4350;
        font-family: Metropolis;
        font-size: 22px;
        font-weight: 700;
        line-height: 28px;
    }

    .enterprise-license-text {
        max-height: 200px;
        padding: 10px;
        background: white;
        overflow-y: auto;
        text-align: justify;
    }

    .content-footer {
        display: flex;

        button {
            border-radius: 4px;
            margin-top: 15px !important;
            margin-left: auto;
        }
    }

    .modal-footer {
        border-top: none !important;
    }
}
