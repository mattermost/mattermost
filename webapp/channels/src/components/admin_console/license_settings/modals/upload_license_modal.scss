.UploadLicenseModal {
    .modal-header {
        border-bottom: none !important;
        background: none !important;

        button.close {
            span {
                color: rgba(63, 67, 80, 0.75) !important;
            }
        }
    }

    .modal-body {
        padding-top: 0 !important;
        padding-bottom: 24px !important;
    }

    .content-body {
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;

        .svg-image {
            margin-bottom:15px;
        }

        .hands-svg {
            margin-top: 0;
            margin-bottom: 15px;
        }

        .title {
            margin: 10px 0;
            color: #3f4350;
            font-family: Metropolis;
            font-size: 22px;
            font-weight: 700;
            line-height: 28px;
        }

        .subtitle {
            width: 90%;
            color: #3f4350;
            font-size: 14px;
            font-style: normal;
            font-weight: normal;
            line-height: 20px;
            text-align: center;
        }

        .file-upload {
            width: 90%;
            padding: 12px;
            border: 1px solid rgba(63, 67, 80, 0.32);
            border-radius: 4px;
            margin-top: 15px;

            &__titleSection {
                color: rgba(63, 67, 80, 0.75);
                font-size: 12px;
                font-weight: bold;
                line-height: 16px;
            }

            &__inputSection {
                display: flex;
                justify-content: space-between;
                margin-top: 10px;

                .file-name-section {
                    display: flex;

                    svg {
                        margin-right: 5px;
                    }

                    .file-size {
                        margin-left: 20px;
                    }
                }

                .file__upload {
                    position: relative;
                    display: flex;
                    align-self: flex-end;
                    margin-left: auto;

                    a {
                        color: #1c58d9;
                        font-weight: 600;
                    }

                    input {
                        position: absolute;
                        z-index: 5;
                        top: 0;
                        left: 0;
                        width: 100%;
                        height: 100%;
                        padding-left: 100%;
                        cursor: pointer;

                        &[disabled] {
                            cursor: not-allowed;
                        }
                    }
                }
            }
        }

        .serverError {
            display: flex;
            width: 90%;
            margin-top: 5px;
            color: var(--error-text);
            float: left;

            i {
                display: inline;
                margin-right: 5px;
            }

            span.server-error-text {
                display: inline-flex;
            }
        }
    }

    .content-footer {
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        margin-top: 20px;

        button {
            height: 48px;
            padding: 5px 24px;
            border: 0 !important;
            border-radius: 4px;
            margin-top: 15px !important;
            font-size: 16px;
            font-weight: 600;
            line-height: 18px;
        }
    }

    .modal-footer {
        border-top: none !important;
        background: none !important;
    }
}
