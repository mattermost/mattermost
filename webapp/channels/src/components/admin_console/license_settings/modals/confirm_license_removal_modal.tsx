// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {FormattedMessage} from 'react-intl';
import {useSelector, useDispatch} from 'react-redux';

import {GenericModal} from '@mattermost/components';

import {closeModal} from 'actions/views/modals';
import {isModalOpen} from 'selectors/views/modals';

import AlertSvg from 'components/common/svg_images_components/alert_svg';

import {ModalIdentifiers} from 'utils/constants';

import type {GlobalState} from 'types/store';

import './confirm_license_removal_modal.scss';

type Props = {
    currentLicenseSKU: string;
    onExited?: () => void;
    handleRemove?: (e: React.MouseEvent<HTMLButtonElement>) => Promise<void>;
}

const ConfirmLicenseRemovalModal: React.FC<Props> = (props: Props): JSX.Element | null => {
    const dispatch = useDispatch();

    const show = useSelector((state: GlobalState) => isModalOpen(state, ModalIdentifiers.CONFIRM_LICENSE_REMOVAL));
    if (!show) {
        return null;
    }

    const handleOnClose = () => {
        if (props.onExited) {
            props.onExited();
        }
        dispatch(closeModal(ModalIdentifiers.CONFIRM_LICENSE_REMOVAL));
    };

    const handleRemoval = (e: React.MouseEvent<HTMLButtonElement>) => {
        if (props.handleRemove) {
            props.handleRemove(e);
        }
        dispatch(closeModal(ModalIdentifiers.CONFIRM_LICENSE_REMOVAL));
    };

    return (
        <GenericModal
            compassDesign={true}
            className={'ConfirmLicenseRemovalModal'}
            show={show}
            id='ConfirmLicenseRemovalModal'
            onExited={handleOnClose}
        >
            <>
                <div className='content-body'>
                    <div className='alert-svg'>
                        <AlertSvg
                            width={130.5}
                            height={103.5}
                        />
                    </div>
                    <div className='title'>
                        <FormattedMessage
                            id='admin.license.confirm-license-removal.title'
                            defaultMessage='Are you sure?'
                        />
                    </div>
                    <div className='subtitle'>
                        <FormattedMessage
                            id='admin.license.confirm-license-removal.subtitle'
                            defaultMessage='Removing the license will downgrade your server from {currentSKU} to Free. You may lose information. '
                            values={{currentSKU: props.currentLicenseSKU}}
                        />
                    </div>
                </div>
                <div className='content-footer'>
                    <button
                        onClick={handleOnClose}
                        className='btn light-blue-btn'
                        id='cancel-removal'
                    >
                        <FormattedMessage
                            id='admin.license.confirm-license-removal.cancel'
                            defaultMessage='Cancel'
                        />
                    </button>
                    <button
                        onClick={handleRemoval}
                        className='btn btn-primary'
                        id='confirm-removal'
                    >
                        <FormattedMessage
                            id='admin.license.confirm-license-removal.confirm'
                            defaultMessage='Confirm'
                        />
                    </button>
                </div>
            </>
        </GenericModal>
    );
};

export default ConfirmLicenseRemovalModal;
