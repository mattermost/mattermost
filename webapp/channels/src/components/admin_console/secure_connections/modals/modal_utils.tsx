// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {useState} from 'react';
import {useDispatch} from 'react-redux';

import type {Channel} from '@mattermost/types/channels';
import type {StatusOK} from '@mattermost/types/client4';
import type {ServerError} from '@mattermost/types/errors';
import type {RemoteClusterPatch, RemoteCluster, RemoteClusterAcceptInvite} from '@mattermost/types/remote_clusters';
import type {PartialExcept} from '@mattermost/types/utilities';

import {Client4} from 'mattermost-redux/client';

import {openModal} from 'actions/views/modals';

import {ModalIdentifiers} from 'utils/constants';
import {cleanUpUrlable} from 'utils/url';

import SecureConnectionAcceptInviteModal from './secure_connection_accept_invite_modal';
import SecureConnectionCreateInviteModal from './secure_connection_create_invite_modal';
import SecureConnectionDeleteModal from './secure_connection_delete_modal';
import SharedChannelsAddModal from './shared_channels_add_modal';
import SharedChannelsRemoveModal from './shared_channels_remove_modal';

import type {TLoadingState} from '../utils';

const chars = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz~_!@-#$^';
const makePassword = () => {
    return Array.from(window.crypto.getRandomValues(new Uint32Array(16))).
        map((n) => chars[n % chars.length]).
        join('');
};

export const useRemoteClusterCreate = () => {
    const dispatch = useDispatch();
    const [saving, setSaving] = useState<TLoadingState>(false);

    const promptCreate = (patch: RemoteClusterPatch) => {
        return new Promise<RemoteCluster | undefined>((resolve, reject) => {
            dispatch(openModal({
                modalId: ModalIdentifiers.SECURE_CONNECTION_CREATE_INVITE,
                dialogType: SecureConnectionCreateInviteModal,
                dialogProps: {
                    creating: true,
                    onConfirm: async () => {
                        try {
                            setSaving(true);
                            const response = await Client4.createRemoteCluster({
                                ...patch,
                                name: cleanUpUrlable(patch.display_name),
                            });
                            setSaving(false);

                            if (response) {
                                const {invite, password, remote_cluster: remoteCluster} = response;

                                resolve(remoteCluster);
                                return {remoteCluster, share: {invite, password}};
                            }
                        } catch (err) {
                            // handle create error
                            reject(err);
                        }
                        setSaving(false);
                        return undefined;
                    },
                },
            }));
        });
    };

    return {promptCreate, saving};
};

export const useRemoteClusterCreateInvite = (remoteCluster: RemoteCluster) => {
    const dispatch = useDispatch();
    const [saving, setSaving] = useState<TLoadingState>(false);

    const promptCreateInvite = () => {
        return new Promise<RemoteCluster>((resolve, reject) => {
            dispatch(openModal({
                modalId: ModalIdentifiers.SECURE_CONNECTION_CREATE_INVITE,
                dialogType: SecureConnectionCreateInviteModal,
                dialogProps: {
                    onConfirm: async () => {
                        try {
                            const password = makePassword();
                            setSaving(true);
                            const invite = await Client4.generateInviteRemoteCluster(remoteCluster.remote_id, {password});
                            setSaving(false);
                            resolve(remoteCluster);
                            return {remoteCluster, share: {invite, password}};
                        } catch (err) {
                            // handle create error
                            reject(err);
                        }
                        setSaving(false);
                        return undefined;
                    },
                },
            }));
        });
    };

    return {promptCreateInvite, saving} as const;
};

export const useRemoteClusterAcceptInvite = () => {
    const dispatch = useDispatch();
    const [saving, setSaving] = useState<TLoadingState>(false);

    const promptAcceptInvite = () => {
        return new Promise<RemoteCluster>((resolve, reject) => {
            dispatch(openModal({
                modalId: ModalIdentifiers.SECURE_CONNECTION_ACCEPT_INVITE,
                dialogType: SecureConnectionAcceptInviteModal,
                dialogProps: {
                    onConfirm: async (acceptInvite: PartialExcept<RemoteClusterAcceptInvite, 'display_name' | 'default_team_id' | 'invite' | 'password'>) => {
                        try {
                            setSaving(true);
                            const rc = await Client4.acceptInviteRemoteCluster({
                                ...acceptInvite,
                                name: cleanUpUrlable(acceptInvite.display_name),
                            });
                            setSaving(false);
                            resolve(rc);
                            return rc;
                        } catch (err) {
                            // handle create error
                            reject(err);
                            setSaving(err);
                            throw (err);
                        }
                    },
                },
            }));
        });
    };

    return {promptAcceptInvite, saving} as const;
};

export const useRemoteClusterDelete = (rc: RemoteCluster) => {
    const dispatch = useDispatch();
    const promptDelete = () => {
        return new Promise((resolve, reject) => {
            dispatch(openModal({
                modalId: ModalIdentifiers.SECURE_CONNECTION_DELETE,
                dialogType: SecureConnectionDeleteModal,
                dialogProps: {
                    displayName: rc.display_name,
                    onConfirm: () => Client4.deleteRemoteCluster(rc.remote_id).then(resolve, reject),
                },
            }));
        });
    };

    return {promptDelete} as const;
};

export const useSharedChannelsRemove = (remoteId: string) => {
    const dispatch = useDispatch();
    const promptRemove = (channelId: string) => {
        return new Promise((resolve, reject) => {
            dispatch(openModal({
                modalId: ModalIdentifiers.SHARED_CHANNEL_REMOTE_UNINVITE,
                dialogType: SharedChannelsRemoveModal,
                dialogProps: {
                    onConfirm: () => Client4.sharedChannelRemoteUninvite(remoteId, channelId).then(resolve, reject),
                },
            }));
        });
    };

    return {promptRemove};
};

export type SharedChannelsAddResult = {
    data: {[channel_id: string]: PromiseSettledResult<StatusOK>};
    errors: {[channel_id: string]: ServerError};
}
export const useSharedChannelsAdd = (remoteId: string) => {
    const dispatch = useDispatch();
    const promptAdd = () => {
        return new Promise((resolve) => {
            dispatch(openModal({
                modalId: ModalIdentifiers.SHARED_CHANNEL_REMOTE_INVITE,
                dialogType: SharedChannelsAddModal,
                dialogProps: {
                    remoteId,
                    onConfirm: async (channels: Channel[]) => {
                        const result: SharedChannelsAddResult = {data: {}, errors: {}};
                        const {data, errors} = result;

                        const requests = channels.map(({id}) => Client4.sharedChannelRemoteInvite(remoteId, id));
                        (await Promise.allSettled(requests)).forEach((r, i) => {
                            if (r.status === 'rejected' && r.reason.server_error_id) {
                                errors[channels[i].id] = r.reason;
                            } else if (r.status === 'fulfilled') {
                                data[channels[i].id] = r;
                            }
                        });

                        resolve(result);
                        return result;
                    },
                },
            }));
        });
    };

    return {promptAdd};
};
