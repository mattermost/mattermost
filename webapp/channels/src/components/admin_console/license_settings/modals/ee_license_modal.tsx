// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {useSelector, useDispatch} from 'react-redux';

import {GenericModal} from '@mattermost/components';

import {closeModal} from 'actions/views/modals';
import {isModalOpen} from 'selectors/views/modals';

import {ModalIdentifiers} from 'utils/constants';

import type {GlobalState} from 'types/store';

import './ee_license_modal.scss';

type Props = {
    onClose?: () => void;
}

const EELicenseModal: React.FC<Props> = (props: Props): JSX.Element | null => {
    const dispatch = useDispatch();

    const show = useSelector((state: GlobalState) => isModalOpen(state, ModalIdentifiers.ENTERPRISE_EDITION_LICENSE));
    if (!show) {
        return null;
    }

    const handleOnClose = () => {
        if (props.onClose) {
            props.onClose();
        }
        dispatch(closeModal(ModalIdentifiers.ENTERPRISE_EDITION_LICENSE));
    };

    // Note: DO NOT LOCALISE THESE STRINGS. Legally we can not since the license is in English.
    return (
        <GenericModal
            modalHeaderText={'Enterprise Edition License'}
            footerContent={<div/>}
            compassDesign={true}
            className={'EELicenseModal'}
            show={show}
            id='EELicenseModal'
            onExited={handleOnClose}
            tabIndex={-1}
        >
            <>
                <div className='enterprise-license-text'>
                    <div>
                        <p>{'The Mattermost Enterprise Edition (EE) license (the “EE License”)'}</p>
                        <p>{'Copyright (c) 2016-present Mattermost, Inc.'}</p>
                        <p>{'The subscription-only features of the Mattermost Enterprise Edition software and associated documentation files (the "Software") may only be used if you (and any entity that you represent) (i) have agreed to, and are in compliance with, the Mattermost Subscription Terms of Service, available at https://mattermost.com/enterprise-edition-terms/ (the “EE Terms”), and (ii) otherwise have a valid Mattermost Enterprise Edition subscription for the correct features, number of user seats and instances of Mattermost Enterprise Edition that you are running, accessing, or using.  You may, however, utilize the free version of the Software (with several features not enabled) under this license without a license key or subscription provided that you otherwise comply with the terms and conditions of this Agreement. Subject to the foregoing, except as explicitly permitted in the EE Terms, it is forbidden to copy, merge, modify, publish, distribute, sublicense, stream, perform, display, create derivative works of and/or sell the Software in either source or executable form without written agreement from Mattermost.  Notwithstanding anything to the contrary, free versions of the Software are provided “AS-IS” without indemnification, support, or warranties of any kind, expressed or implied. You assume all risk associated with any use of free versions of the Software.'}</p>
                        <p>{'EXCEPT AS OTHERWISE SET FORTH IN A BINDING WRITTEN AGREEMENT BETWEEN YOU AND MATTERMOST, THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.'}</p>
                    </div>
                </div>
            </>
        </GenericModal>
    );
};

export default EELicenseModal;
