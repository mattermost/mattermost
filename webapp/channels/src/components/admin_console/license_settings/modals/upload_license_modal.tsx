// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import marked from 'marked';
import React, {useRef} from 'react';
import {defineMessage, FormattedDate, FormattedMessage} from 'react-intl';
import {useSelector, useDispatch} from 'react-redux';

import {GenericModal} from '@mattermost/components';
import type {ClientLicense} from '@mattermost/types/config';

import {uploadLicense} from 'mattermost-redux/actions/admin';
import {getLicenseConfig} from 'mattermost-redux/actions/general';
import {getLicense} from 'mattermost-redux/selectors/entities/general';

import {closeModal} from 'actions/views/modals';
import {getCurrentLocale} from 'selectors/i18n';
import {isModalOpen} from 'selectors/views/modals';

import FileSvg from 'components/common/svg_images_components/file_svg';
import SuccessSvg from 'components/common/svg_images_components/success_svg';
import UploadLicenseSvg from 'components/common/svg_images_components/upload_license';
import LoadingWrapper from 'components/widgets/loading/loading_wrapper';

import {FileTypes, ModalIdentifiers} from 'utils/constants';
import {getMonthLong} from 'utils/i18n';
import {getSkuDisplayName} from 'utils/subscription';
import {fileSizeToString} from 'utils/utils';

import type {GlobalState} from 'types/store';

import './upload_license_modal.scss';

type Props = {
    onExited?: () => void;
    fileObjFromProps: File | null;
}

const UploadLicenseModal = (props: Props): JSX.Element | null => {
    const dispatch = useDispatch();

    const [fileObj, setFileObj] = React.useState<File | null>(props.fileObjFromProps);
    const [isUploading, setIsUploading] = React.useState(false);
    const [serverError, setServerError] = React.useState<string | null>(null);
    const [uploadSuccessful, setUploadSuccessful] = React.useState(false);
    const fileInputRef = useRef<HTMLInputElement>(null);

    const currentLicense: ClientLicense = useSelector(getLicense);
    const locale = useSelector(getCurrentLocale);

    const handleChange = () => {
        const element = fileInputRef.current;
        if (element === null || element.files === null || element.files.length === 0 || element.files[0].size === 0) {
            return;
        }
        setFileObj(element.files[0]);
        setServerError(null);
    };

    const handleSubmit = async (e: React.MouseEvent<HTMLButtonElement>) => {
        e.preventDefault();
        if (fileObj === null) {
            return;
        }

        setIsUploading(true);
        const {error} = await dispatch(uploadLicense(fileObj));

        if (error) {
            setFileObj(null);
            setServerError(error.message);
            setIsUploading(false);
            return;
        }

        await dispatch(getLicenseConfig());
        setFileObj(null);
        setServerError(null);
        setIsUploading(false);
        setUploadSuccessful(true);
    };

    const show = useSelector((state: GlobalState) => isModalOpen(state, ModalIdentifiers.UPLOAD_LICENSE));
    if (!show) {
        return null;
    }

    const handleOnClose = () => {
        if (isUploading) {
            return;
        }
        if (props.onExited) {
            props.onExited();
        }
        dispatch(closeModal(ModalIdentifiers.UPLOAD_LICENSE));
    };

    const handleRemoveFile = () => {
        setFileObj(null);
    };

    const displayFileName = (fileName: string) => {
        const extLen = FileTypes.LICENSE_EXTENSION.length;
        let fileNameWithoutExt = fileName.split(FileTypes.LICENSE_EXTENSION)[0];
        fileNameWithoutExt = fileNameWithoutExt.length < (40 - extLen) ? fileNameWithoutExt : `${fileNameWithoutExt.substr(0, (37 - extLen))}...`;
        return `${fileNameWithoutExt}${FileTypes.LICENSE_EXTENSION}`;
    };

    let uploadLicenseContent = (
        <>
            <div className='content-body'>
                <div className='svg-image'>
                    <UploadLicenseSvg
                        width={151}
                        height={103}
                    />
                </div>
                <div className='title'>
                    <FormattedMessage
                        id='admin.license.upload-modal.title'
                        defaultMessage='Upload a License Key'
                    />
                </div>
                <div className='subtitle'>
                    <FormattedMessage
                        id='admin.license.upload-modal.subtitle'
                        defaultMessage='Upload a license key for Mattermost Enterprise Edition to upgrade this server. '
                    />
                </div>
                <div className='file-upload'>
                    <div className='file-upload__titleSection'>
                        <FormattedMessage
                            id='admin.license.upload-modal.file'
                            defaultMessage='File'
                        />
                    </div>
                    <div className='file-upload__inputSection'>
                        <div className='help-text file-name-section'>
                            {fileObj?.name && fileObj?.size ? (
                                <>
                                    <FileSvg
                                        width={20}
                                        height={20}
                                    />
                                    <span className='file-name'>
                                        {displayFileName(fileObj.name)}
                                    </span>
                                    <span className='file-size'>
                                        {fileSizeToString(fileObj.size)}
                                    </span>
                                </>
                            ) : (
                                <FormattedMessage
                                    id='admin.license.no-file-selected'
                                    defaultMessage='No file selected'
                                />
                            )}
                        </div>
                        <div className='file__upload'>
                            {fileObj?.name ? (
                                <a
                                    onClick={handleRemoveFile}
                                >
                                    <FormattedMessage
                                        id='admin.license.remove'
                                        defaultMessage='Remove'
                                    />
                                </a>
                            ) : (
                                <>
                                    <input
                                        ref={fileInputRef}
                                        type='file'
                                        accept={FileTypes.LICENSE_EXTENSION}
                                        onChange={handleChange}
                                    />
                                    <a
                                        className='btn-select'
                                    >
                                        <FormattedMessage
                                            id='admin.license.choose'
                                            defaultMessage='Choose File'
                                        />
                                    </a>
                                </>
                            )}
                        </div>
                    </div>
                </div>
                {serverError && <div className='serverError'>
                    <i className='icon icon-alert-outline'/>
                    <span
                        className='server-error-text'
                        dangerouslySetInnerHTML={{__html: marked(serverError)}}
                    />
                </div>}
            </div>
            <div className='content-footer'>
                <div className='btn-upload-wrapper'>
                    <button
                        className={`btn ${(fileObj?.name && fileObj?.name.length > 0) && 'btn-primary'}`}
                        disabled={!(fileObj?.name && fileObj?.name.length > 0)}
                        onClick={handleSubmit}
                        id='upload-button'
                    >
                        <LoadingWrapper
                            loading={Boolean(isUploading)}
                            text={defineMessage({id: 'admin.license.modal.uploading', defaultMessage: 'Uploading'})}
                        >
                            <FormattedMessage
                                id='admin.license.modal.upload'
                                defaultMessage='Upload'
                            />
                        </LoadingWrapper>
                    </button>
                </div>
            </div>
        </>
    );

    if (uploadSuccessful) {
        const startsAt = (
            <FormattedDate
                value={new Date(parseInt(currentLicense.StartsAt, 10))}
                day='2-digit'
                month={getMonthLong(locale)}
                year='numeric'
            />
        );
        const expiresAt = (
            <FormattedDate
                value={new Date(parseInt(currentLicense.ExpiresAt, 10))}
                day='2-digit'
                month={getMonthLong(locale)}
                year='numeric'
            />
        );

        const licensedUsersNum = currentLicense.Users;
        const skuName = getSkuDisplayName(currentLicense.SkuShortName, currentLicense.IsGovSku === 'true');
        uploadLicenseContent = (
            <>
                <div className='content-body'>
                    <div className='svg-image hands-svg'>
                        <SuccessSvg
                            width={162}
                            height={103.5}
                        />
                    </div>
                    <div className='title'>
                        <FormattedMessage
                            id='admin.license.upload-modal.successfulUpgrade'
                            defaultMessage='Successful Upgrade!'
                        />
                    </div>
                    <div className='subtitle'>
                        <FormattedMessage
                            id='admin.license.upload-modal.successfulUpgradeText'
                            defaultMessage='You have upgraded to the {skuName} plan for {licensedUsersNum, number} seats. This is effective from {startsAt} until {expiresAt}. '
                            values={{
                                expiresAt,
                                startsAt,
                                licensedUsersNum,
                                skuName,
                            }}
                        />
                    </div>
                </div>
                <div className='content-footer'>
                    <div className='btn-upload-wrapper'>
                        <button
                            className='btn btn-primary'
                            onClick={handleOnClose}
                            id='done-button'
                        >
                            <FormattedMessage
                                id='admin.license.modal.done'
                                defaultMessage='Done'
                            />
                        </button>
                    </div>
                </div>
            </>
        );
    }

    return (
        <GenericModal
            className={'UploadLicenseModal'}
            show={show}
            id='UploadLicenseModal'
            compassDesign={true}
            onExited={handleOnClose}
        >
            {uploadLicenseContent}
        </GenericModal>
    );
};

export default UploadLicenseModal;
