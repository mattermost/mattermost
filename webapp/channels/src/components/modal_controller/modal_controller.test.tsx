// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {mount} from 'enzyme';
import React from 'react';
import {Modal} from 'react-bootstrap';
import {Provider} from 'react-redux';

import {closeModal} from 'actions/views/modals';

import mockStore from 'tests/test_store';

import ModalController from '.';

type TestModalProps = {
    onExited: () => void;
}

type TestModalState = {
    show: boolean;
}

class TestModal extends React.PureComponent<TestModalProps, TestModalState> {
    constructor(props: TestModalProps) {
        super(props);

        this.state = {
            show: true,
        };
    }

    hideModal = () => {
        this.setState({show: false});
    };

    render() {
        return (
            <Modal
                show={this.state.show}
                onHide={this.hideModal}
                onExited={this.props.onExited}
            >
                <Modal.Header closeButton={true}/>
                <Modal.Body/>
            </Modal>
        );
    }
}

describe('components/ModalController', () => {
    const modalId = 'test_modal';

    test('component should match snapshot without any modals', () => {
        const state = {
            views: {
                modals: {
                    modalState: {},
                },
            },
        };

        const store = mockStore(state);

        const wrapper = mount(
            <Provider store={store}>
                <ModalController/>
            </Provider>,
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('ModalController > *').length).toBe(0);
        expect(document.getElementsByClassName('modal-dialog').length).toBeFalsy();
    });

    test('test model should be open', () => {
        const state = {
            views: {
                modals: {
                    modalState: {
                        [modalId]: {
                            open: true,
                            dialogProps: {},
                            dialogType: TestModal,
                        },
                    },
                },
            },
        };

        const store = mockStore(state);

        mount(
            <Provider store={store}>
                <ModalController/>
            </Provider>,
        );

        expect(document.getElementsByClassName('modal-dialog').length).toBe(1);
    });

    test('should pass onExited to modal to allow a modal to remove itself', () => {
        const state = {
            views: {
                modals: {
                    modalState: {
                        [modalId]: {
                            open: true,
                            dialogProps: {},
                            dialogType: TestModal,
                        },
                    },
                },
            },
        };

        const store = mockStore(state);

        const wrapper = mount(
            <Provider store={store}>
                <ModalController/>
            </Provider>,
        );

        expect(wrapper.find(TestModal).exists()).toBe(true);
        expect(wrapper.find(TestModal).prop('onExited')).toBeDefined();
        expect(wrapper.find(Modal).prop('onExited')).toBeDefined();

        wrapper.find(TestModal).prop('onExited')!();

        expect(store.getActions()).toEqual([
            closeModal(modalId),
        ]);
    });

    test('should call a provided onExited in addition to removing the modal', () => {
        const onExited = jest.fn();

        const state = {
            views: {
                modals: {
                    modalState: {
                        [modalId]: {
                            open: true,
                            dialogProps: {
                                onExited,
                            },
                            dialogType: TestModal,
                        },
                    },
                },
            },
        };

        const store = mockStore(state);

        const wrapper = mount(
            <Provider store={store}>
                <ModalController/>
            </Provider>,
        );

        expect(wrapper.find(TestModal).exists()).toBe(true);
        expect(wrapper.find(TestModal).prop('onExited')).toBeDefined();
        expect(wrapper.find(Modal).prop('onExited')).toBeDefined();

        expect(onExited).not.toHaveBeenCalled();

        wrapper.find(TestModal).prop('onExited')!();

        expect(onExited).toHaveBeenCalled();
    });
});
