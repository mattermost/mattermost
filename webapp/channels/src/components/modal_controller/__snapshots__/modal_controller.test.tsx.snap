// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`components/ModalController component should match snapshot without any modals 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ModalController)>
    <ModalController
      actions={
        Object {
          "closeModal": [Function],
        }
      }
      modals={
        Object {
          "modalState": Object {},
        }
      }
    />
  </Connect(ModalController)>
</Provider>
`;
